File: src/Tools/Dova.Tools.Java.ClassGenerator/src/main/java/com/github/sejoslaw/dova/ClassDetailsDefinitionModel.java
package com.github.sejoslaw.dova;

import java.util.ArrayList;
import java.util.Collection;

public class ClassDetailsDefinitionModel {
    public String PackageName;
    public String ClassName;
    public boolean IsEnum;
    public String Modifiers;
    public Collection<TypeParameterModel> TypeParameterModels = new ArrayList<>();
    public String Signature;
}

File: src/Tools/Dova.Tools.JavaClassStructureGenerator/Models/ClassDetailsDefinitionModel.cs
namespace Dova.Tools.JavaClassStructureGenerator.Models;

internal class ClassDetailsDefinitionModel
{
    public string PackageName { get; set; }
    public string ClassName { get; set; }
    public bool IsEnum { get; set; }
    public string Modifiers { get; set; }
    public IEnumerable<TypeParameterModel> TypeParameterModels { get; set; } = new List<TypeParameterModel>();
    public string Signature { get; set; }
}